### PR TITLE
Add ifEmpty operators on Maybe trait

### PR DESCRIPTION
**Pull request by pietrocaselani**
This PR adds two operators on `Maybe`:
* ifEmpty(default:)
* ifEmpty(switchTo:)

Both operators are available on [RxJava Maybe](http://reactivex.io/RxJava/javadoc/io/reactivex/Maybe.html)

### CODE REVIEW

**Comment by pietrocaselani** [comment]
Any update on this?
Do I need to update the Changelog?

File: RxSwift/Traits/Maybe.swift [hunk 1]
```diff
@@ -227,4 +227,40 @@ public extension PrimitiveSequenceType where TraitType == MaybeTrait {
...
+
+    /**
+     Emits elements from the source observable sequence, or a default element if the source observable sequence is empty.
+
+     - seealso: [DefaultIfEmpty operator on reactivex.io](http://reactivex.io/documentation/operators/defaultifempty.html)
+
+     - parameter default: Default element to be sent if the source does not emit any elements
+     - returns: An observable sequence which emits default element end completes in case the original sequence is empty
+     */
+    public func ifEmpty(default: ElementType) -> Maybe<ElementType> {
```
Comment (kzaher) [inline RxSwift/Traits/Maybe.swift]: Shouldn't this return `Single<ElementType>`?

File: Tests/RxSwiftTests/MaybeTest.swift [hunk 2]
```diff
@@ -585,6 +585,49 @@ extension MaybeTest {
             completed(200)
             ])
     }
+
+    func test_ifEmptyDefault() {
+        let scheduler = TestScheduler(initialClock: 0)
+
+        let res = scheduler.start {
+            (Maybe<Int>.empty().ifEmpty(default: 5).asObservable())
```
Comment (kzaher) [inline Tests/RxSwiftTests/MaybeTest.swift]: This should be of form:
```
    (Maybe<Int>.empty().ifEmpty(default: 5) as Single<Int>).asObservable()
```

File: Tests/RxSwiftTests/MaybeTest.swift [hunk 3]
```diff
@@ -585,6 +585,49 @@ extension MaybeTest {
...
+            ])
+    }
+
+    func test_ifEmptySwitchToMaybe() {
+        let scheduler = TestScheduler(initialClock: 0)
+        let source = Maybe<Int>.empty()
+        let switchSource = Maybe.just(10)
+
+        let res = scheduler.start {
+            (source.ifEmpty(switchTo: switchSource).asObservable())
```
Comment (kzaher) [inline Tests/RxSwiftTests/MaybeTest.swift]: I think this should be:
```
(Maybe<Int>.empty().ifEmpty(switchTo: Maybe.just(10)) as Maybe<Int>).asObservable()
```

File: Tests/RxSwiftTests/MaybeTest.swift [hunk 4]
```diff
@@ -585,6 +585,49 @@ extension MaybeTest {
...
+            ])
+    }
+
+    func test_ifEmptySwitchToSingle() {
+        let scheduler = TestScheduler(initialClock: 0)
+        let source = Maybe<Int>.empty()
+        let switchSource = Single.just(10)
+
+        let res = scheduler.start {
+            (source.ifEmpty(switchTo: switchSource).asObservable())
```
Comment (kzaher) [inline Tests/RxSwiftTests/MaybeTest.swift]:
```
(Maybe<Int>.empty().ifEmpty(switchTo: Single.just(10)) as Single<Int>).asObservable()
```

Comment (kzaher) [reply Tests/RxSwiftTests/MaybeTest.swift]: The reason is because we want to test return type. That's one of the critical parts.

**Pull request merged**